cinema/views.py

```diff
@@ -24,28 +27,60 @@
 class GenreViewSet(viewsets.ModelViewSet):
     queryset = Genre.objects.all()
     serializer_class = GenreSerializer
+    authentication_classes = (TokenAuthentication,)
+
+    def get_permissions(self) -> list:
+        if self.action in ("list", "create"):
+            permission_classes = [IsAdminOrIfAuthenticatedReadOnly]
+        else:
+            raise NotFound()
```

Comment:
In the `get_permissions` method, raising a `NotFound` exception for actions other than 'list' and 'create' might not be appropriate. Consider using `MethodNotAllowed` for unsupported actions to provide a more accurate response.